x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java

```diff
@@ -62,13 +65,15 @@ public static void deletePolicy(String name, ClusterService clusterService, Cons
             throw new IllegalArgumentException("name is missing or empty");
         }
 
-        final Map<String, EnrichPolicy> policies = getPolicies(clusterService.state());
-        if (policies.containsKey(name) == false) {
-            throw new ResourceNotFoundException("policy [{}] not found", name);
-        }
+        updateClusterState(clusterService, handler, current -> {
+            final Map<String, EnrichPolicy> policies = getPolicies(current);
```

Comment:
should the retrieval and throw happen outside of the task? The throw will just bubble up as an exception in the caller of the `EnrichStore` whereas before it was a thrown exception.

Reply:
We have to check whether the policy exists in the update task, only here we can be certain about checks. We can also check it outside the update task, before we update the cluster state, but then we would check it twice. 

The error will be caught by ClusterService and passed down to the handler (via ClusterStateUpdateTask's onFailure() method), so I think we're good here?